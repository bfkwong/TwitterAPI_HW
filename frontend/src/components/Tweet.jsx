import { Grid, Typography, IconButton } from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import React from "react";
import { deleteTweet } from "../Requests";

export default function Tweet(props) {
  return (
    <Grid container style={{ borderStyle: "solid", borderWidth: 0, borderBottomWidth: 1, paddingBottom: 15 }}>
      <Grid item xs>
        <Typography variant="body1">{props?.tweet?.text}</Typography>
        <Typography variant="subtitle2">
          {new Date(props?.tweet?.created_at).toLocaleString()} by @{props?.tweet?.user?.screen_name}
        </Typography>
      </Grid>
      <Grid item>
        <IconButton
          id={`dlt_tweet_${props?.tweet?.id_str}`}
          onClick={async () => {
            await deleteTweet(props?.tweet?.id_str);
            props.onDelete();
          }}>
          <DeleteIcon />
        </IconButton>
      </Grid>
    </Grid>
  );
}
