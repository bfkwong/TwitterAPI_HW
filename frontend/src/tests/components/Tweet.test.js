import Tweet from "../../components/Tweet";
import { fireEvent, render } from "@testing-library/react";

jest.mock("../../Requests.js", () => ({
  getAllTweets: () => true,
  postTweet: () => true,
  deleteTweet: () => true,
  authenticate: () => true
}));

test("Tweet with no props should render without failure", () => {
  const component = render(<Tweet />);
  expect(component).toMatchSnapshot();
});

test("Tweet with prop should show correctly", async () => {
  const onDeleteMock = jest.fn();
  const component = render(
    <Tweet
      onDelete={onDeleteMock}
      tweet={{ text: "Hello world", created_at: "9/3/2021", id_str: "12345", user: { screen_name: "@person" } }}
    />
  );
  expect(onDeleteMock.mock.calls.length).toBe(0);

  const deleteTweetBtn = component.container.querySelector("#dlt_tweet_12345");
  expect(deleteTweetBtn).toBeTruthy();
  await fireEvent.click(deleteTweetBtn);

  expect(onDeleteMock.mock.calls.length).toBe(1);

  expect(component).toMatchSnapshot();
});
