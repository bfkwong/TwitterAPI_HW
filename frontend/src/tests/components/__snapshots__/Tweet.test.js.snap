// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet with no props should render without failure 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="border-style: solid; border-width: 0px; border-bottom-width: 1px; padding-bottom: 15px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
          <h6
            class="MuiTypography-root MuiTypography-subtitle2"
          >
            Invalid Date
             by @
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            id="dlt_tweet_undefined"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container"
      style="border-style: solid; border-width: 0px; border-bottom-width: 1px; padding-bottom: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        />
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          Invalid Date
           by @
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          id="dlt_tweet_undefined"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tweet with prop should show correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="border-style: solid; border-width: 0px; border-bottom-width: 1px; padding-bottom: 15px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Hello world
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2"
          >
            9/3/2021, 12:00:00 AM
             by @
            @person
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            id="dlt_tweet_12345"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container"
      style="border-style: solid; border-width: 0px; border-bottom-width: 1px; padding-bottom: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Hello world
        </p>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          9/3/2021, 12:00:00 AM
           by @
          @person
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          id="dlt_tweet_12345"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
